src/main/java/io/confluent/connect/jdbc/source/JdbcSourceConnectorConfig.java

```diff
@@ -252,6 +258,31 @@ public boolean visible(String name, Map<String, Object> config) {
     }
   }
 
+  /**
+   * A recommender for table names that caches the table names for the last configuration.
+   */
+  private static class CachingTableRecommender extends TableRecommender {
+
+    private Map<String, Object> lastConfig;
+    private List<Object> results;
+
+    @Override
+    public List<Object> validValues(String name, Map<String, Object> config) {
+      if (lastConfig != null && lastConfig.equals(config)) {
```

Comment:
Shouldn't this have a timeout or something as well? Even if the config doesn't change, the DB might...

Reply:
yeah, that would work too. i think we have this pattern most places because generally the configdef is immutable

Reply:
Making the caching recommender be threadsafe and have an expiration should be enough. Just about to push a second commit that makes those changes, makes the table recommender static again (albeit it's now a caching recommender wrapper), and adds quite a few more tests. Thanks for the feedback.